micro_dl/utils/mp_utils.py

```diff
@@ -47,53 +48,54 @@ def create_save_mask(input_fnames,
     :param int int2str_len: Length of str when converting ints
     :param str mask_type: thresholding type used for masking or str to map to
      masking function
+    :param str mask_ext: 'npy' or 'png'. Save the mask as uint8 PNG or
+         NPY files
     :return dict cur_meta for each mask
     """
 
-    im_stack = np.squeeze(tile_utils.read_imstack(
+    im_stack = tile_utils.read_imstack(
         input_fnames,
         flat_field_fname,
         normalize_im=False,
-    ))
-    # Combine channel images and generate mask
-    if len(im_stack.shape) == 2:
-        summed_image = im_stack
-    elif len(im_stack.shape) == 3:
-        if len(input_fnames) == 1:
-            # read a 3d image
-            summed_image = im_stack
-        else:
-            # read a 2d image stack
-            summed_image = np.sum(np.stack(im_stack), axis=2)
-    elif len(im_stack.shape) == 4:
-        # read a 3d image stack
-        summed_image = np.sum(np.stack(im_stack), axis=3)
-    summed_image = summed_image.astype('float32')
-    if mask_type == 'otsu':
-        mask = mask_utils.create_otsu_mask(summed_image, str_elem_radius)
-    elif mask_type == 'unimodal':
-        mask = mask_utils.create_unimodal_mask(summed_image,
-                                               str_elem_radius)
+    )
+    masks = []
+    for idx in range(im_stack.shape[-1]):
```

Comment:
I guess this changes the behavior of masking? Instead of adding images together then masking the summed image you add the generated masks together?
At the very least I think the multi-channel behavior needs to be clarified in the docstring.

Reply:
Yes, summing images first doesn't work well when the intensities of channels are not on similar scale. I documented the behavior in the docstring